code/game/objects/items/devices/multitool.dm

```diff
@@ -41,14 +41,25 @@
 	user.visible_message(span_suicide("[user] puts the [src] to [user.p_their()] chest. It looks like [user.p_theyre()] trying to pulse [user.p_their()] heart off!"))
 	return OXYLOSS//theres a reason it wasn't recommended by doctors
 
+/**
+ * Sets the multitool internal object buffer
+ *
+ * Arguments:
+ * * buffer - the new object to assign to the multitool's buffer
+ */
 /obj/item/multitool/proc/set_buffer(datum/buffer)
 	if(src.buffer)
 		UnregisterSignal(src.buffer, COMSIG_QDELETING)
-	if(QDELETED(buffer))
-		return
 	src.buffer = buffer
-	RegisterSignal(buffer, COMSIG_QDELETING, PROC_REF(on_buffer_del))
+	if(!QDELETED(buffer))
+		RegisterSignal(buffer, COMSIG_QDELETING, PROC_REF(on_buffer_del))
```

Comment:
Why?

Reply:
That's not possible because a lot of places pass null to `set_buffer()` for e.g

https://github.com/tgstation/tgstation/blob/17562248c90410d445f1213994e5aad498205d36/code/game/machinery/computer/mechlaunchpad.dm#L103

QDELETED checks for null as well. Creating a seperate proc just to set buffer to null is just excess work at this point